irohad/subscription/thread_handler.hpp

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_SUBSCRIPTION_THREAD_HANDLER_HPP
+#define IROHA_SUBSCRIPTION_THREAD_HANDLER_HPP
+
+#include <assert.h>
+#include <chrono>
+#include <deque>
+#include <functional>
+#include <mutex>
+#include <shared_mutex>
+#include <thread>
+
+#include "subscription/common.hpp"
+
+namespace iroha::subscription {
+
+  class threadHandler final : utils::NoCopy, utils::NoMove {
+   public:
+    using Task = std::function<void()>;
+
+   private:
+    using Time = std::chrono::high_resolution_clock;
+    using Timepoint = std::chrono::time_point<Time>;
+    using TimedTask = std::pair<Timepoint, Task>;
+    using TaskContainer = std::deque<TimedTask>;
+
+    std::atomic_flag proceed_;
+    mutable std::mutex tasks_cs_;
```

Comment:
what means `cs` in variable name? 
```suggestion
    mutable std::mutex mutex_;
```